### PR TITLE
Bf product controller

### PR DESCRIPTION
**Pull request by CreativeFitz**
# Description

I added a class for Product as well as a controller and a test to check the ProductController.

## Related Ticket(s)
Ticket 2

## Steps to Test
After you pull down branch bf-product-controller, you should be able to begin test from ProductTest.cs which will check for proper GET all products, GET a product, PUT and DELETE calls. All six test should pass. Delete functionality is only soft deleting by changing Archived key to true instead of permanently deleting from DB. Only permanent delete are those Products created in the processing of tests.

# Checklist:
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new errors
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: BangazonAPI/BangazonAPI/Controllers/ProductController.cs [hunk 1]
```diff
@@ -29,37 +31,248 @@ public SqlConnection Connection
...
+                    conn.Open();
+                    using (SqlCommand cmd = conn.CreateCommand())
+                    {
+                        if (delete == true)
+                        {
+                            cmd.CommandText = @"DELETE FROM Product WHERE Id = @id";
+                        }
+                        else
+                        {
+                            cmd.CommandText = @"UPDATE PaymentType SET Archived=1 WHERE Id =@id";
```
Comment (kimpreece) [inline BangazonAPI/BangazonAPI/Controllers/ProductController.cs]: Don't delete PaymentType

**Review by kimpreece — CHANGES_REQUESTED** [review]
See comment on the delete.

File: BangazonAPI/BangazonAPI/Controllers/ProductController.cs (hunk 1 above)
Comment (CreativeFitz) [reply BangazonAPI/BangazonAPI/Controllers/ProductController.cs]: Should be fixed now Kim

Comment (CreativeFitz) [reply BangazonAPI/BangazonAPI/Controllers/ProductController.cs]: Now code shall update soft delete change on get method.

Comment (CreativeFitz) [reply BangazonAPI/BangazonAPI/Controllers/ProductController.cs]: Now code shall update soft delete change on get method.

**Comment by kimpreece** [comment]
Looks good! You're good to merge!

**Pull request merged**